youtube_dl/extractor/iqiyi.py

```diff
@@ -301,10 +302,14 @@ def get_raw_data(self, tvid, video_id):
     def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
-        links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
-            webpage)
-        if not links:
+        links = []
+        for link in re.findall(r'<a[^>]+class="[^"]*site-piclist_pic_link[^"]*"[^>]*>', webpage):
```

Comment:
@yan12125 Any hints about how I should modify get_element_by_class without breaking anything?
a) Add a new parameter to get_element_by_class() 'multiple=false' which changes the behavior of get_element_by_class?
b) Add a new function get_element_by_class_multiple() / get_elements_by_class() ? (thats no big change because get_element_by_class() itself relies solely on get_element_by_attribute.)

I have searched through the code but I don't see any example function which returns only one element and changes it's behavior to returning multiple when the caller wants it.

Of course there is the third option of simply changing get_element_by_class() to return multiple resulsts, but:
1. We have to make sure that all calls to it are working fine afterwards (about 9 calls).
2. The name of the function is then wrong because it returns not only one result anymore.

And when I'm up to it, what's with:
- get_element_by_id()
- get_element_by_attribute()

All of them can return multiple resulsts :-(